pkg/io/jsonio/unmarshal.go

```diff
@@ -269,15 +350,50 @@ func unmarshal(filename string) (*spec.Spec, error) {
 	return jsonSpec, err
 }
 
-func cidrContainedInVpc(cidr ipblock.IPBlock, addressPrefixes []ir.CIDR) (bool, error) {
-	for i := range addressPrefixes {
-		addressPrefix, err := ipblock.FromCidr(addressPrefixes[i].String())
-		if err != nil {
-			return false, err
+func parseOverlappingVpcs(cidr ipblock.IPBlock, vpcs map[ir.ID]*ir.VPCDetails) ([]ir.ID, error) {
+	result := make([]ir.ID, 0)
+	for vpcName, vpcDetails := range vpcs {
+		for _, addressPrefix := range vpcDetails.AddressPrefixes {
+			addressPrefixIP, err := ipblock.FromCidr(addressPrefix.String())
+			if err != nil {
+				return nil, err
+			}
+			if !addressPrefixIP.Intersect(&cidr).IsEmpty() {
+				result = append(result, vpcName)
+			}
 		}
-		if cidr.ContainedIn(addressPrefix) {
-			return true, nil
+	}
+	return result, nil
+}
+
+func replaceResourceName(cache map[string]ir.ID, ambiguous map[string]struct{}, resourceName string,
+	resourceType spec.ResourceType) (string, error) {
+	if len(ir.ScopingComponents(resourceName)) != 1 {
+		return resourceName, nil
+	}
+	if val, ok := cache[resourceName]; ok {
+		return val, nil
+	}
+	if _, ok := ambiguous[resourceName]; ok {
+		return "", fmt.Errorf("ambiguity error for resource named %s", resourceName)
+	}
+	return "", fmt.Errorf("unknown resource type %q named %s", resourceType, resourceName)
```

Comment:
```suggestion
	return "", fmt.Errorf("unknown resource name %s (resource type: %q)", resourceName, resourceType)
```
It's the name that is unknown, not the type